tcms/management/admin.py

```diff
@@ -34,6 +35,12 @@ class ComponentAdmin(admin.ModelAdmin):
     def get_queryset(self, request):
         return super().get_queryset(request).select_related('product', 'initial_owner')
 
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if db_field.name == "initial_owner" or db_field.name == "initial_qa_contact":
+            kwargs["queryset"] = get_user_model().objects.order_by('username')
+
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
```

Comment:
Drop this one!

Although it works this is not a good solution. Ideally we would like to provide default ordering for the User model everywhere not only on this particular admin page. 

OTOH having the user admin page sorted by ID is nice b/c it gives you the latest users on top.

There's no one size fits all solution for the user model and there are conflicting requirements so better remove this.

I prefer having cleaner and more consistent code instead of patch here and there solution.